pkg/kubecodegen/pin.go

```diff
@@ -0,0 +1,10 @@
+// +build tools
+
+// Dummy package to pin kubernetes/code-generator
+// from go.mod used by 'make kube-gen'.
+// Required to allow "go mod vendor" to fetch the same kubecodegen
```

Comment:
This feels like we're misusing go.mod.

Maybe we can trust quarks-utils to pick the right version of codegen? If it's a concern, that quarks-utils should work with any controller-runtime (in case quarks-operator and quarks-job drift apart), I imagine the helpers in quarks-utils  are already tightly coupled to the controller-runtime version?

Reply:
That is the 'best practice' suggested by go mod authors: see https://github.com/golang/go/issues/25624#issuecomment-395556484 https://github.com/golang/go/issues/25922
.
 You can also see that practice is being used in controller-runtime as well.. https://github.com/kubernetes-sigs/controller-runtime/blob/eec4f7489bf469a7dcf261cc79bab698d8521533/hack/tools/tools.go#L19 and in the official crd example : https://github.com/kubernetes/sample-controller/blob/bf5c061a2037b0c8a0da885e3cca780c92f23503/hack/tools.go#L22 (which this set of prs is inspired from )

Apparently this is the approach being used everywhere.. :shrug:

Reply:
@mudler why are we pinning the code-generator here? shouldn't quarks-utils do that, and  then we pin the utils?

Reply:
Here we pin to the version that is refered in quarks-utils: https://github.com/cloudfoundry-incubator/quarks-utils/pull/45 

Otherwise, the single projects could pull a different version.. in this way we make sure to pull the same one that is pinned in quarks-utils

Reply:
See https://github.com/cloudfoundry-incubator/quarks-utils/pull/45#discussion_r422842836 
In the go.mod here indeed we don't pin kubegen, but we just refer to quarks-utils version